data_model/src/lib.rs

```diff
@@ -391,6 +470,12 @@ impl Value {
     }
 }
 
+impl From<BlockValue> for Value {
```

Comment:
Consider `derive_more::From` with a type annotation.

Reply:
`FromVariant` is redundant. But I agree, that we can remove it in one fell swoop later.